GN: update to .com/ads csp rules

fatimaarhab opened the pull request:
https://jira.dev.bbc.co.uk/browse/GNADTECH-2189

**Overall change:**
This is a small PR to tidy up CSP rules for advertising and enable the feature Funding Choices CMP domain too

On test from US proxy: 
www.test.bbc.com/mundo?fundingchoices_cmp
www.test.bbc.com/mundo?ias_publisher

**Code changes:**

- tidy up CSP rules for advertising

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

Commit:
GN: update to .com/ads csp rules

Commit:
Merge branch 'latest' into GN_CSP_Domain_Update

andrewscfc commented on src/server/utilities/cspHeader/index.js:
```diff
@@ -12,64 +12,41 @@ import { bbcDomains, advertisingServiceCountryDomains } from './domainLists';
-    'https://detect-survey.effectivemeasure.net',
+    'https://*.googlesyndication.com',
+    'https://*.doubleclick.net',
+    'https://*.effectivemeasure.net',
```
Looks like we've got quite a few wildcard csp entries being introduced here, is this 'tidying up' or strictly necessary, it would better to keep the specific entries as they were if the currently serve the required purpose.

fatimaarhab replied:
I thought it'd simplify things, and make the csp file smaller... I can go back to fix entry...

andrewscfc replied:
It technically make the csp less secure as it allows more domains, as a guideline wildcard usage in csps should be minimized. I agree it's more readable and tidy but technically less secure unfortunately

andrewscfc commented on src/server/utilities/cspHeader/index.js:
```diff
@@ -12,64 +12,41 @@ import { bbcDomains, advertisingServiceCountryDomains } from './domainLists';
-    'https://ad.doubleclick.net',
-    'https://googleads.g.doubleclick.net',
-    'https://securepubads.g.doubleclick.net',
+    'fundingchoicesmessages.google.com',
```
I'm not completely certain but I suspect you need `https://` adding to the start of these domains for this change to work properly?

bbc-fatima replied:
https://experience.tinypass.com is not in use on WS sites
*crwdcntrl.net that's a third party that has been deprecated

I've also removed some instances like these
'https://googleads.g.doubleclick.net',
'https://securepubads.g.doubleclick.net'
As they are already covered by 'https://*.g.doubleclick.net',
Ideally we'd have all the separate instances but that's a tough one to get :(

Hope this helps

Added
'https://fundingchoicesmessages.google.com', // new CMP launching soon
    'https://secure-dcr-cert.imrworldwide.com', // something that is currently missing
    'https://pixel.adsafeprotected.com', // new ad safety launching soon
    'https://cdn.adsafeprotected.com', // new ad safety launching soon

andrewscfc commented on src/server/utilities/cspHeader/index.js:
```diff
@@ -12,64 +12,41 @@ import { bbcDomains, advertisingServiceCountryDomains } from './domainLists';
+    'https://*.googlesyndication.com',
+    'https://*.google.com',
+    'https://*.adsafeprotected.com',
+    '*.imrworldwide.com',
```
```suggestion
    'https://*.imrworldwide.com',
```
probably?

andrewscfc reviewed (commented):
I have dropped a few comments at the beginning of the PR but they apply throughout, I hope they are helpful.

Commit:
Update src/server/utilities/cspHeader/index.js

Co-authored-by: Andrew Bennett <andrew.bennett07@bbc.co.uk>

Commit:
GN: update to .com/ads csp rules
remove simplifications

Commit:
Merge branch 'latest' into GN_CSP_Domain_Update

Commit:
Merge branch 'latest' into GN_CSP_Domain_Update

andrewscfc reviewed (approved):
LGTM.

Just to be clear you have removed some domains @bbc-fatima was this part of your tidy-up? Just if this change was purely to aid a new release there would be no removals I imagine?

fatimaarhab commented:
> LGTM.
> 
> Just to be clear you have removed some domains @bbc-fatima was this part of your tidy-up? Just if this change was purely to aid a new release there would be no removals I imagine?

https://experience.tinypass.com is not in use on WS sites
*crwdcntrl.net that's a third party that has been deprecated

I've also removed some instances like these
'https://googleads.g.doubleclick.net',
'https://securepubads.g.doubleclick.net'
As they are already covered by 'https://*.g.doubleclick.net',
Ideally we'd have all the separate instances but that's a tough one to get :(

Hope this helps

Added
'https://fundingchoicesmessages.google.com', // new CMP launching soon
    'https://secure-dcr-cert.imrworldwide.com', // something that is currently missing
    'https://pixel.adsafeprotected.com', // new ad safety launching soon
    'https://cdn.adsafeprotected.com', // new ad safety launching soon
